src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1042,10 +1042,69 @@ Node* VectorNode::try_to_gen_masked_vector(PhaseGVN* gvn, Node* node, const Type
   }
 }
 
+bool VectorNode::should_swap_inputs() {
+  // Must be a binary operation.
+  if (req() != 3) {
+    return false;
+  }
+
+  // Predicated vector operations are sensitive to ordering of inputs.
+  // When the mask corresponding to a vector lane is false then
+  // the result of the operation is corresponding lane of its first operand.
+  //   i.e. RES = VEC1.lanewise(OPER, VEC2, MASK) is semantically equivalent to
+  //        RES = BLEND(VEC1, VEC1.lanewise(OPER, VEC2), MASK)
+  if (is_predicated_vector()) {
+    return false;
+  }
```

Comment:
Hmm. Can you give me a concrete example of a masked operation that would be filtered out?

Can it for example be a `AddVI`? But that only has 2 inputs for `VEC1` and `VEC2`. Where would the mask be located - and why does that not get us to `req() > 3`?

Ah, I see it can be added in `VectorNode::try_to_gen_masked_vector`, with `add_req`, but then we should have `req() > 3`.

Ok, this looks a bit complicated, but it looks like we are doing this.
```
  // Generate a vector mask for vector operation whose vector length is lower than the
  // hardware supported max vector length.
```

Ok, fine.

It could be good to add a comment here though, explaining why the operation seemingly has 3 inputs, but we don't exit at `req() != 3` above.

Reply:
Hi @eme64, Since we are selectively enabling commoning based on opcodes checks, hence skipping over their predicated counterparts should suffice.

Please find below the code reference adding mask operands it targets supports perdicated instructions.
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectorIntrinsics.cpp#L495